x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -216,15 +216,17 @@ protected Query asQuery(IsNotNull isNotNull, TranslatorHandler handler) {
         }
 
         public static Query doTranslate(IsNotNull isNotNull, TranslatorHandler handler) {
-            Query query = null;
+            return handler.wrapFunctionQuery(isNotNull, isNotNull.field(), () -> translate(isNotNull, handler));
+        }
 
+        private static Query translate(IsNotNull isNotNull, TranslatorHandler handler) {
```

Comment:
Use a different name to avoid confusion, it's counter intuitive for `doTranslate` to call `translate`, I would have expected it to be the other way around.
Maybe `createQuery` or `internalTranslate`

Reply:
+1

Reply:
Might be a tad confusing, but this "doTranslate -> translate" pattern seems to be a [pre-existing](https://github.com/elastic/elasticsearch/blob/ae94701db6e82d8aae98f751f214869922bf34e4/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java#L268) one.

Reply:
As @bpintea pointed out the `doTranslate / translate` methods are pre-existing. I would do the renames in a separate PR across the all the translators and keep this PR focused only on laziness.

Reply:
Also fine for me, to address this later!